EmployeeLeaveTracking/EmployeeLeaveTracking.Services/Services/LeaveRequestService.cs

```diff
@@ -378,11 +413,42 @@ public async Task<int> UpdateLeaveRequestStatus(int id, int statusId)
                 await _dbContext.SaveChangesAsync();
             }
 
+            User employeeEmail = await _dbContext.Users.FindAsync(leaveRequest.EmployeeId);
+
+            if (employeeEmail == null)
+            {
+                throw new Exception("Employee not found.");
+            }
+
+            // using the appropriate email template based on the statusId
+            string emailTemplateKey = (statusId == 2) ? "LeaveApprovedTemplate" : "LeaveRejectedTemplate";
+
+            //accessing the resource file
+            ResourceManager resourceManager = new ResourceManager("EmailService.EmailTemplates", typeof(EmailHelper).Assembly);
+
+            //retrieving the email template from the resource file
+            string emailTemplate = resourceManager.GetString(emailTemplateKey);
+
+            //replace placeholders with appropriate values
+            string leaveStatus = (statusId == 2) ? "Approved" : "Rejected";
+            string emailBody = string.Format(emailTemplate,
+                leaveRequest.StartDate.ToShortDateString(),
+                leaveRequest.EndDate.ToShortDateString(),
+                leaveRequest.TotalDays,
+                leaveRequest.RequestComments,
+                leaveStatus);
+            //return manager and employee names with this response to display in email template
+
+            EmailHelper emailService = new EmailHelper();
+            string emailToAddress = "rajgurupreeti0@gmail.com";
```

Comment:
Move to config's

Reply:
It is actually implemented using : 
string emailToAddress = employeeEmail.Email;
Instead of : 
string emailToAddress = "rajgurupreeti0@gmail.com"; (Had added this hardcoded email for testing purpose)
Therefore can't be used as configurable value